### ISSUE TITLE
Crashes when resuming from suspend mode

### ISSUE THREAD

**Issue by kekeblom** [microsoft/vscode#97688]
Issue Type: <b>Bug</b>

Every time I wake up my system from suspend mode, vscode will use 100% of a single core and freeze indefinitely. The projects that was open was a cmake c++ project. I only had one tab/one file open and nothing crazy.

I'm running Ubuntu 18.04 with Nvidia drivers 440.82. 

VS Code version: Code 1.45.0 (d69a79b73808559a91206d73d7717ff5f798f23c, 2020-05-07T16:10:24.511Z)
OS version: Linux x64 5.3.0-51-generic

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-8550U CPU @ 1.80GHz (8 x 3159)|
|GPU Status|2d_canvas: enabled<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>oop_rasterization: disabled_off<br>protected_video_decode: unavailable_off<br>rasterization: disabled_software<br>skia_renderer: disabled_off_ok<br>video_decode: unavailable_off<br>viz_display_compositor: enabled_on<br>viz_hit_test_surface_layer: disabled_off_ok<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|3, 2, 2|
|Memory (System)|15.37GB (0.58GB free)|
|Process Argv|--no-sandbox --unity-launch|
|Screen Reader|no|
|VM|0%|
|DESKTOP_SESSION|ubuntu|
|XDG_CURRENT_DESKTOP|Unity|
|XDG_SESSION_DESKTOP|ubuntu|
|XDG_SESSION_TYPE|x11|
</details><details><summary>Extensions (4)</summary>

Extension|Author (truncated)|Version
---|---|---
latex-workshop|Jam|8.9.0
cmake-tools|ms-|1.3.1
cpptools|ms-|0.28.0
vim|vsc|1.14.2

</details>

**Comment by alexdima** [comment]
@kekeblom It would be very helpful for us to know which process is using 100% CPU. Can you please run `code --status` or share with us the full command line of the misbehaving process so we get some kind of idea what could be causing this?

**Comment by kekeblom** [comment]
`code --status` just freezes and never returns.

**Comment by alexdima** [comment]
I think that might be Chromium's GPU process. I think the solution is to run with `--disable-gpu`, but let's ask @deepak1556 if he knows something more.

**Comment by deepak1556** [comment]
Would be good to try the workaround mentioned by @alexdima first, `--disable-gpu` only disables hw acceleration but the gpu process is also responsible for software mode, so it may or may not resolve the issue but will hint where the problem lies.

The ideal next step is to capture a chrome trace and see the calls from gpu process, but we only currently support for capturing startup traces using `--trace` flag in VSCode. Let me fix this to allow arbitrary captures and we can proceed from there.

Can you also try to provide a minimal repro that would trigger this (I believe this has to involve the cmake-tools extension) ? Thanks!

**Comment by kekeblom** [comment]
`--disable-gpu` does the trick. If I run `code --disable-extensions .` to start VSCode, suspend and wake up the system, it freezes again. This would lead me to believe it isn't related to the extensions I have installed.

Let me know if there's any other info I can provide.

**Comment by deepak1556** [comment]
Thanks for testing!

> Let me know if there's any other info I can provide.

I would still like a minimal repro, like a sample c++ project that could trigger this.

* Can you also check if the exploration build works fine with hw acceleration  https://github.com/microsoft/vscode/issues/94074#issuecomment-607079209

* Finally, was there a recent driver update for your gpu ? Any possibility to roll back and verify ?

**Comment by kekeblom** [comment]
It seems is has nothing to do with c++ or the project. I can reproduce this by running the following steps:
```
mkdir empty_dir && cd empty_dir
code --disable-extensions .
sudo pm-suspend
```
Then wake up the system and it's frozen.

It also freezes when I don't use the `--disable-extensions` flag.

My guess would be that issue is somehow related to the kernel and Nvidia driver version. 

Unfortunately, I need cuda on this system for work stuff and given that Nvidia drivers on Ubuntu are a house of cards, I would rather not mess around with my current configuration. Especially when I have finally got the system to a mostly stable state, besides this issue, which gladly I can now workaround.

**Comment by kekeblom** [comment]
The exploration build does not seem to suffer from this issue. :+1: :tada:

**Comment by johnbendi** [comment]
@deepak1556 Thank God I found this thread. I have been suffering this problem for some months now. Initially it started by cancelling the scaling on a hidpi laptop running Ubuntu Linux 19.10 and later 20.04. The problem starts as reported by @kekeblom 

On 19.10 both Chrome browser and VSCode looses scaling at the same time and VSCode hangs but Chrome does not. I end up having to kill VSCode and Chrome and restart to get things back to normal. And that's like everytime I foget to kill VSCode before suspending.

On 20.04 it no longer looses the scaling and only VSCode hangs with the process mentioned by @kekeblom 

@kekeblom please how do I try the exploration build

**Comment by kekeblom** [comment]
As deepka1556 mentioned, you can download the deb from here https://github.com/microsoft/vscode/issues/94074#issuecomment-607079209

The deb file can then be installed using `sudo dpkg -i <path-to-deb-file>`. You should now be able to launch the app through `code-exploration` or through the launch menu by searching for code exploration.

**Comment by TheShed412** [comment]
I am suffering from a similar issue. VScode doesn't fully crash for me, the scrolling code viewer on the right gets scrambled and the terminal also becomes scrambled.

![vscode_scramble](https://user-images.githubusercontent.com/9973218/87623966-8f7ee780-c6f4-11ea-8a1b-b0a80f8afa52.png)

The issue seems to lie with the Nvidia drivers. I recently installed 440.100 for doing stuff with Vulkan when I noticed this. When I switch back to the xorg drivers, the issue goes away. I haven't tried the deb @deepak1556 suggested yet, so I might try it to see if it remedies my issue.

When waking from suspend with the nvidia drivers, I was able to see some ACPI errors that popped up right before the login screen started.

![110301360_714828089078620_1836123804329387014_n](https://user-images.githubusercontent.com/9973218/87624417-bbe73380-c6f5-11ea-9244-7830f9ca697b.png)

Those didn't show up when I was waking from suspend with the xorg driver. 

Not sure if this helps at all but I came here to submit a ticket and this one seemed related.
